actors/builtin/paych/paych_test.go

```diff
@@ -0,0 +1,179 @@
+package paych_test
+
+import (
+	"context"
+	"testing"
+
+	addr "github.com/filecoin-project/go-address"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	. "github.com/filecoin-project/specs-actors/actors/builtin/paych"
+	"github.com/filecoin-project/specs-actors/actors/crypto"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestPaymentChannelActor_Constructor(t *testing.T) {
+	ctx := context.Background()
+	actor := pcActorHarness{Actor{}, t}
+
+	pcaAddr := tutil.NewIDAddr(t, 100)
+	callerAddr := tutil.NewIDAddr(t, 101)
+
+	t.Run("can create a payment channel actor", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, pcaAddr).
+			WithCaller(callerAddr, builtin.AccountActorCodeID).
```

Comment:
This immediately reveals problems with the actor. The actor's constructor must be changed to only be callable by the InitActor (like all other actors). I think that means that the constructor params must be expanded to explicitly include the `From` address, which will no longer be the immediate caller.

Reply:
You can then factor this builder setup out of each t.Run, because it will always be the same.